TOOLS/odoo-tools.sh

```diff
@@ -1224,23 +1227,124 @@ if [ "$SCRIPT_MODE" = "updatetranslation" ]; then
+    if ! [ $# -ge 3 ] || [ $# -gt 4 ]; then
+        echo -e "ERROR: \"setup-toosl.sh $SCRIPT_MODE\" takes a minimum of three and a maximum of four arguments!"
         exit 2
     fi
 
-    # TODO: check or create INSTANCE/DATABASE/BACKUP folder
-    # Todo: CD to BACKUP folder
-    # Todo: create backup with dbname--DATE.zip with DATE in Format 2014-12-30--24-59
+    # ----- Initialize Variables
+    DBNAME=$3
+    TARGET_BRANCH=$2
+    if ! [ "x$4" = "x" ]; then
+        if [ $4 == "odoozip" ] || [ $4 == "etherpad" ] || [ $4 == "owncloud" ] || [ $4 == "full" ]; then
+            TYPE=$4
+        else
+            echo "ERROR: ${4} must be empty or in [odoozip|etherpad|owncloud|full] !"; exit 2
+        fi
+    else
+        TYPE="odoozip"
+    fi
 
+    # ----- Global Variables
+    DATETIME=`date +%Y-%m-%d__%H-%M`
+    BRANCH_PATH="${REPOPATH}/${TARGET_BRANCH}"
+    BRANCHLOGFILE="${REPOPATH}/SETUP/IS-BACKUP--${DATETIME}.log"
+    INSTANCE_PATH="${REPOPATH}/${TARGET_BRANCH}"
 
+    # ----- Find Odoo-Instance(s) to Backup
+    if [ ${DBNAME} = "all" ]; then
+        GREPREGEX="\bo8_[0-9a-zA-Z]*_[0-9a-zA-Z]*"
+    else
+        GREPREGEX="\b${DBNAME}"
+    fi
+    echo "DEBUG: regex: ${GREPREGEX}, dbname: ${DBNAME}"
+    # Todo: Use the correct linux user instead of SU except for full backup. Check rights of psql
+    declare -a DATABASES=($(su - postgres -c "psql --tuples-only -P format=unaligned -c \"SELECT datname FROM pg_database WHERE datname LIKE 'o8_%'\""|grep -o ${GREPREGEX}|sort -u))
+    if [ "x${DATABASES}" = "x" ]; then
+        echo "ERROR: No Database found!"
+    fi
+
+    # ----- HINT: Spezial case STOP Script till odoo Supports this option, service/db.py needs to be extended
+    if [ ${TYPE} = "odoosql" ]; then
+        echo "ERROR: This option is not supported right now "
+        exit 2
+    fi
+    # Todo: check vmware Snapshot how to remote execute the vmware-cmd command with ssh connection to esx server directly, check if the VM is running on this machine
+    # Todo: or find a way of acting through Virtual center server this server has access to the whole cluster and no check on which host a machine is running would be needed
+    # ----- Backup Data for each Instance
+    counter=3
+    for i in "${DATABASES[@]}"; do
+        # Check if we are at PAD CLOUD OR ODOO DB
+
+        # ----- Getting config of database an Parameters
+        INSTANCELOGFILE="${INSTANCE_PATH}/${i}/LOG/IS-BACKUP--${i}--${DATETIME}.log"
+        echo "instancelogfile: ${INSTANCELOGFILE}"
+        INSTANCECONFIGFILE=${INSTANCE_PATH}/${i}/${i}.conf
+        echo "Database Config File --> ${INSTANCECONFIGFILE}" | tee -a ${INSTANCELOGFILE} ${BRANCHLOGFILE}
+        BASEPORT69=($(grep "xmlrpc_port" ${INSTANCECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+        SUPER_PASSWORD=($(grep "admin_passwd" ${INSTANCECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+        BACKUPFILE=${INSTANCE_PATH}/${i}/BACKUP/IS-BACKUP--${i}--${DATETIME}
```

Comment:
Backupfile Naming convention should be set here - important for restore also - personally i would suggest that we follow this naming convention e.g.:
Odoo: 
o8_intdadirl1_ahch-odoo-2015_01_23_2330.zip
Pad:
o8_intdadirl1_ahch-pad-2015_01_23_2345.zsql
Cloud:
o8_intdadirl1_ahch-cloud-2015_01_24_0005.zip

with the following convention for file endings:
.sql = plain text sql dump
.zsql = plain sql dump zipped
.dump = custom format sql dump
.zdump = custom format gziped sql dump
.zip = A zip archive (may contain also sql dumps of course)